course3/homework-unique_ptr/test.cc

```diff
@@ -0,0 +1,99 @@
+#include "unique_ptr.h"
+#include "unique_ptr.cc"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+
+
+template<template<class...> class UniquePtrType>
```

Comment:
True; template-templates are fancy and complex. You need to have good reason to use them. In this case a valid strategy would be to a) not make this a template function, simply test your pointer only, b) assume fixed pointed-to type:
```
template<typename PtrType>
void unique_ptr_test()
{
// ...
}

// ...
unique_ptr_test<std::unique_ptr<int>>();
unique_ptr_test<karun::unique_ptr<int>>();
```